compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -202,14 +217,6 @@ impl<'tcx> Inliner<'tcx> {
 
         if let Some(callee_def_id) = callee.def_id().as_local() {
             let callee_hir_id = self.tcx.hir().local_def_id_to_hir_id(callee_def_id);
-            // Avoid inlining into generators,
-            // since their `optimized_mir` is used for layout computation, which can
-            // create a cycle, even when no attempt is made to inline the function
-            // in the other direction.
-            if caller_body.generator.is_some() {
```

Comment:
The cycle can still happen by inlining `Pin::<&mut from_generator::GenFuture<[generator]>>::new_unchecked`, which is defined in a parent crate.